Frontend/android-native/mobility-common/src/main/java/in/juspay/mobility/common/CircleRippleEffect.java

```diff
@@ -38,24 +50,87 @@ public void updatePosition(LatLng center) {
         }
     }
 
-    public void draw(@NonNull GoogleMap googleMap, LatLng center) {
-        ExecutorManager.runOnMainThread(() -> this.circle = googleMap.addCircle(new CircleOptions()
-                .center(center)
-                .fillColor(mOptions.getFillColor())
-                .strokeColor(Color.parseColor(mOptions.getFromStrokeColor()))
-                .strokeWidth(mOptions.getStrokeWidth())
-                .visible(true)
+    public void draw(@NonNull GoogleMap googleMap, LatLng center, BridgeComponents bridgeComponents, String callback) {
+        ExecutorManager.runOnMainThread(() -> {
+            this.circle = googleMap.addCircle(new CircleOptions()
+                    .center(center)
+                    .fillColor(mOptions.getFillColor())
+                    .strokeColor(Color.parseColor(mOptions.getFromStrokeColor()))
+                    .strokeWidth(mOptions.getStrokeWidth())
+                    .visible(true)
+                    .radius(mOptions.getRadius()));
+            updateStrokePatternForCircle(mOptions);
+            if (googleMap != null && mOptions.getIsCircleClickable()) {
```

Comment:
Do we need a null check, since you mentioned it as NonNull? if you still need that, can you wrap the `this.circle = ...` also inside null check.